app/validators.rb/locked_field_validator.rb

```diff
@@ -0,0 +1,18 @@
+class LockedFieldValidator < ActiveModel::Validator
+  def initialize(options)
+    super
+    @locked_value = options[:locked_value]
+    @attribute = options[:attribute]
+  end
+
+  def validate(record)
+    attribute_was = record.send("#{@attribute}_was")
+    return if attribute_was != @locked_value || record.new_record?
+
+    attribute_changed = record.send("#{@attribute}_changed?")
+    message = "You cannot change the name of the \"#{@constant}\" " \
```

Comment:
D'oh, that's a typo - I changed `@constant` to `@locked_value` but obviously missed that, ta!